File: frontend/app/models/ticket.ts
import { Reader } from "./reader";

export class Ticket{
    ID!: number;
    ReaderID!: number;
    DateStart!: Date;
    DateEnd!: Date;
    Reader!: Reader;

    constructor(id: number, readerID: number, dateStart: Date, dateEnd: Date, reader: Reader){
        this.ID = id;
        this.ReaderID = readerID;
        this.DateStart = dateStart;
        this.DateEnd = dateEnd;
        this.Reader = reader;
    }
}

File: backend/Models/Ticket.cs
using System.ComponentModel.DataAnnotations;

namespace lab8.Models
{
    public class Ticket
    {
        [Key]
        public int ID { get; set; }
        public int ReaderID { get; set; }

        public DateTime DateStart { get; set; }

        public DateTime DateEnd { get; set; }

        public Reader Reader { get; set; } = null!;
    }
}
